Runtime/Tracking/Collision/CollisionNotifier.cs

```diff
@@ -107,7 +107,7 @@ public override bool Equals(object obj)
             /// <inheritdoc />
             public override int GetHashCode()
             {
-                return ColliderData.GetContainingTransform().GetHashCode();
+                return ColliderData.GetHashCode() + (ForwardSource != null ? ForwardSource.GetHashCode() : 0);

```

Comment:
```suggestion
                return ((ForwardSource != null ? ForwardSource.GetHashCode() : 0) * 397) ^ (ColliderData != null ? ColliderData.GetHashCode() : 0);
```
(I used ReSharper's auto-generation for this.)